File: Frontend/manga-world/src/Model/AuthorManga.ts
import Author from "./Author";
import Manga from "./Manga";

class AuthorManga {
  aid: number;
  mid: number;
  author: Author;
  manga: Manga;

  constructor(aid: number, mid: number, author: Author, manga: Manga) {
    this.aid = aid;
    this.mid = mid;
    this.author = author;
    this.manga = manga;
  }
}

export default AuthorManga;

File: Backend/manga-world/src/main/java/com/amakakeru/mangaworld/Entity/AuthorManga.java
package com.amakakeru.mangaworld.Entity;

import com.amakakeru.mangaworld.Entity.IdClasses.AuthorMangaId;
import jakarta.persistence.*;
import lombok.Data;

import java.util.Objects;

@Data
@Entity
@IdClass(AuthorMangaId.class)
@Table(name = "author_manga")
public class AuthorManga {

    @Id
//    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "a_id")
    private Long aid;

    @Id
//    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "m_id")
    private Long mid;

    @ManyToOne
    @JoinColumn(name = "a_id", insertable = false, updatable = false)
    private Author author;

    @ManyToOne
    @JoinColumn(name = "m_id", insertable = false, updatable = false)
    private Manga manga;

    @Override
    public int hashCode() {
        return Objects.hash(aid, mid);
    }

    @Override
    public String toString() {
        return "AuthorManga{" +
                "aid=" + aid +
                ", mid=" + mid +
                '}';
    }
}
